src/Core/Utils/BijectiveAssociation.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <initializer_list>
+#include <map>
+
+namespace Ra {
+namespace Core {
+namespace Utils {
+
+/**
+ * \brief Bijective association on a finite set of <key, value> pairs.
+ * \tparam T1 Type of the key
+ * \tparam T2 Type of the value
+ */
+template <typename T1, typename T2>
+class BijectiveAssociation
+{
+  public:
+    using key_type   = T1;
+    using value_type = T2;
+
+    /**
+     * \brief Constructor from { <T1, T2> } pairs
+     */
+    explicit BijectiveAssociation( std::initializer_list<std::pair<key_type, value_type>> pairs );
+
+    /**
+     * \brief Add a pair to the relation.
+     * \return false if the pair was not added due to collision with an already inserted pair
+     */
+    bool addAssociation( std::pair<key_type, value_type> p );
```

Comment:
I think we could follow std::map naming, e.g. insert instead of addAssociation.

Reply:
I agree.